src/io/EventReader.cc

```diff
@@ -47,6 +47,9 @@ EventReader::result_type EventReader::operator()()
         }
         ++event_id;
 
+        // Track numbering starts at 1 since 0 is reserved for the parent ID
+        int track_id = 1;
```

Comment:
I'm probably misunderstanding how the Geant track numbering works (wouldn't be the first time).. I thought primary particles had parent ID zero, which is why I had primary track IDs starting from 1, but I'm definitely not sure that's right.

Reply:
Oh, you know what I could be misunderstanding as well, I didn't even think of what the *parent* IDs for the "primary" particles are. I guess I assumed they'd be unassigned instead of having a meaningful value.

I'm not sure what exactly the Geant family of codes does, but I think the most sensible thing for us to do is to use track IDs that correspond to the individual tracks that we transport -- 0 through `num_primaries - 1` for the primaries for a particular event. The parent IDs for those I would leave with the "unassigned" ID.

Reply:
Works for me!

Reply:
Per our discussion, Geant4 particle guns start with track ID of 1 as the initial primary, and it has a parent of 0. (This is actually what your first implementation said.) But for events read in externally they can basically have an arbitrary ID. But we are going to do our own thing for now, and make sure that (when needed) we can store the metadata that associates our internal track IDs to HEPEVT-provided IDs.